deployment/services/public-graphql-api-gateway.ts

```diff
@@ -0,0 +1,90 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import * as k8s from '@pulumi/kubernetes';
+import * as pulumi from '@pulumi/pulumi';
+import { ServiceSecret } from '../utils/secrets';
+import { ServiceDeployment } from '../utils/service-deployment';
+import { type Docker } from './docker';
+import { type Environment } from './environment';
+import { type GraphQL } from './graphql';
+
+/**
+ * Hive Gateway Docker Image Version
+ * Bump this to update the used gateway version.
+ */
+const dockerImage = 'ghcr.io/graphql-hive/gateway:1.13.4';
+
+const gatewayConfigDirectory = path.resolve(
+  __dirname,
+  '..',
+  'config',
+  'public-graphql-api-gateway',
+);
+const gatewayConfigPath = path.join(gatewayConfigDirectory, 'gateway.config.ts');
+
+export function deployPublicGraphQLAPIGateway(args: {
+  environment: Environment;
+  graphql: GraphQL;
+  docker: Docker;
+}) {
+  const apiConfig = new pulumi.Config('api');
+
+  // Note: The persisted documents cdn endpoint can also be used for reading the contract schema
+  const cdnEndpoint =
+    apiConfig.requireObject<Record<string, string>>('env')['HIVE_PERSISTED_DOCUMENTS_CDN_ENDPOINT'];
+
+  if (!cdnEndpoint) {
+    throw new Error("Missing cdn endpoint variable 'HIVE_PERSISTED_DOCUMENTS_CDN_ENDPOINT'.");
+  }
+
+  const supergraphEndpoint = cdnEndpoint + '/contracts/public';
+
+  // Note: The persisted documents access key is also valid for reading the supergraph
+  const persistedDocumentsSecret = new ServiceSecret('persisted-documents', {
+    cdnAccessKeyId: apiConfig.requireSecret('hivePersistedDocumentsCdnAccessKeyId'),
+  });
+
+  const configMap = new k8s.core.v1.ConfigMap('public-graphql-api-gateway-config', {
+    data: {
+      'gateway.config.ts': fs.readFileSync(gatewayConfigPath, 'utf-8'),
```

Comment:
nit: I saw in modern examples of Pulumi that they use this instead of reading the file with `fs`: https://www.pulumi.com/registry/packages/local/api-docs/file/ 

Not sure if it's needed, as `fs` worked for us before.